src/lab1/Main.java

```diff
@@ -0,0 +1,179 @@
+        double maxRating = movies[0].getRating();
+        double minRating = movies[0].getRating();
+        for (Movie movie : movies) {
+            if (movie.getRating() > maxRating) {
+                maxRating = movie.getRating();
+            }
+            if (movie.getRating() < minRating) {
+                minRating = movie.getRating();
+            }
+        }
+        System.out.println("The movie with the maximum rating: " + maxRating);
+        System.out.println("The movie with the minimum rating: " + minRating);
+    }
+
+    static void findMovie(String movieName,Movie[] movies) {
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                System.out.println("Rating of movie \"" + movieName + "\" is " + movie.getRating());
+                return;
+            }
+        }
+        System.out.println("Movie \"" + movieName + "\" not found");
+    }
+
+    static void updateMovieRating(String movieName,Movie[] movies,Scanner sc) {
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                System.out.print("Enter the new rating for the movie \"" + movieName + "\": ");
+                movie.setRating(sc.nextDouble());
+                return;
+            }
+        }
+        System.out.println("Movie \"" + movieName + "\" not found");
+    }
+
+    static Movie[] deleteMovie(String movieName,Movie[] movies) {
+        boolean found = false;
+        for (Movie movie : movies) {
+            if (movie.getName().equals(movieName)) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            System.out.println("Movie \"" + movieName + "\" not found");
+            return movies;
+        } else {
+            System.out.println("Movie \"" + movieName + "\" deleted");
+        }
+        Movie[] newMovies = new Movie[movies.length - 1];
+        for(int i = 0, j = 0; i < movies.length; i++, j++) {
+            if(movies[i].getName().equals(movieName)) {
+                j--;
+                continue;
+            }
+            newMovies[j] = movies[i];
+        }
+        return newMovies;
+    }
+
+    static Movie[] sortMovies(Movie[] movies) {
+        for(int i = 0; i < movies.length - 1; i++) {
+            boolean swapped = false;
+            for(int j = 0; j < movies.length - i - 1; j++) {
+                if(movies[j].getRating() > movies[j + 1].getRating()) {
+                    Movie temp = movies[j];
+                    movies[j] = movies[j + 1];
+                    movies[j + 1] = temp;
+                    swapped = true;
+                }
+            }
+            if(!swapped) {
+                return movies;
+            }
+        }
+        return movies;
+    }
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Movie[] movies = new Movie[0];
```

Comment:
instead sending movies and scanner to each method, you can declare these on class level